lib/perl/Genome/Model/Tools/Vcf/EvaluateVcfs.pm

```diff
@@ -0,0 +1,374 @@
+        },
+
+        true_negative_bed => {
+            is => "Text",
+            doc => "BED file containing regions where no "
+                   . "variant call should be made",
+        },
+
+        true_negative_size => {
+            is => "Integer",
+            doc => "Use this number as the size of the TN BED "
+                   . "file rather than calculating on the fly",
+            is_optional => '1',
+        },
+
+        pass_filter_expression => {
+            is => "Text",
+            doc => "String to pass to vcflib vcffilter in order to "
+                   . "select variants from the evaluation VCF",
+            default_value => q{-g 'FT = PASS | FT = .'},
+        },
+    ],
+
+    has_output => [
+        rawdata => {
+            is => "HASH",
+            doc => "The organized stats & metadata generated during "
+                   . "primary execution",
+        },
+    ],
+};
+
+sub execute {
+    my $self = shift;
+    my $configs = $self->parse_config_file();
+    $self->collect_statistics($configs);
+    $self->rawdata($configs);
+    $self->display_summary_stats();
+    return 1;
+}
+
+sub display_summary_stats {
+    my $self = shift;
+    my $configs = $self->rawdata;
+
+    $self->status_message("Generating Summary Report");
+
+    print "\n";
+    print "==================", "\n";
+    print "Valdiation Summary", "\n";
+    print "==================", "\n";
+    print "\n";
+
+    my @stat_names = $self->stat_types;
+    my @headers = ('Name', 'ID', 'VarType', @stat_names);
+    print join("\t", @headers), "\n";
+
+    for my $set (@{$configs}) {
+        my ($name, $id, $type, $stats) =
+          @{$set}{'name', 'id', 'variant_type', 'stats'};
+        my @results = @{$stats}{@stat_names};
+        print join("\t", $name, $id, $type, @results), "\n";
+    }
+}
+
+sub collect_statistics {
+    my ($self, $inputs) = @_;
+
+    my ($i, $total) = (1, scalar @{$inputs});
+    $self->status_message(
+        "Performing Evaluations & Collecting Statistics (%d vcfs)",
+        $total
+    );
+
+    for my $set (@{$inputs}) {
+        $self->status_message(
+            "[ $i | $total ] Name: %s / Model ID: %s / Type: %s",
+            $set->{'name'}, $set->{'id'}, $set->{'variant_type'}
+        );
+        $self->status_message(" ---> Vcf: %s", $set->{'vcf'});
+
+        my $wkspace = $self->create_subdirectory_workspace($set);
+        $self->status_message(" ---> Output Workspace: %s", $wkspace->stringify);
+
+        $self->status_message(" ---> Start VCF Evaluation");
+        my $stats = $self->_run_evaluate_vcf($wkspace, $set);
+        $self->status_message(" ---> Finish VCF Evaluation\n\n");
+
+        $set->{'stats'} = $stats;
+        $i++;
+    }
+}
+
+sub parse_config_file {
+    my $self = shift;
+    my $config = $self->fetch_config_file;
+
+    $self->status_message("Parsing configuration file : $config");
+
+    my $fh = $config->openr;
+
+    my @inputs;
+    while (my $line = <$fh>) {
```

Comment:
If you wanted, you could use TSV reader that's already in `Genome::Utility`

Something along the lines of

```perl
my $reader = Genome::Utility::IO::SeparatedValueReader->create(
  headers => [qw(name id path variant_type sample)],
  input => $self->config_file(),
  separator => "\t"
);

while (my $line_data = $reader->next()) {
   $self->validate_type($line_data->{variant_type});
   ....
}
````
Would eliminate the need for the `fetch_config_file` method as well as the manual `chomp`, `split`, and file handle handling. It would also validate that the file has the correct number of columns for you.